clang/include/clang/Basic/Attr.td

```diff
@@ -1275,6 +1275,20 @@ def SYCLSimdAccessorPtr : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+// Used by FE to mark SYCL kernel pointer parameters which correspond to the
+// original lambda's captured accessors. FE turns the attribute to some metadata
+// required by the FPGA back-end.
+// This attribute does not require custom semantic handling 
+// hence we set the SemaHandler field to 0.
+// The attribute is not for public consumption, and is an implicitly-created attribute
+// that has no visible spelling, hence undocumented.
+def SYCLAccessorPtr : InheritableAttr {
```

Comment:
Should be def SYCLAccessorPtr : Attr?

Reply:
I chose to inherit from `InheritableAttr` based on the below text from the clang [documentaion] :(https://clang.llvm.org/docs/InternalsManual.html#how-to-change-clang):
`The first step to adding a new attribute to Clang is to add its definition to include/clang/Basic/Attr.td. This tablegen definition must derive from the Attr (tablegen, not semantic) type, or one of its derivatives. Most attributes will derive from the InheritableAttr type, which specifies that the attribute can be inherited by later redeclarations of the Decl it is associated with`

Can you explain how does it impact choosing between `Attr` vs `InheritableAtrr `?

Also, for my own learning, can someone clarify when it would be appropriate to use `InheritableParamAttr` ?
It seems to inherit `InheritableAttr`

`InheritableParamAttr is similar to InheritableAttr, except that the attribute is written on a parameter instead of a declaration`

Reply:
I think `Attr `makes sense here since we don't expect redeclaration of openCL kernel. 

Redeclarations are cases like the following - 
```

[[clang::abc]] void foo();
void foo(){} <-- abc is carried over to this declaration as well. 
```

I am not very familiar with `InheritableParamAttr`, since I've never implemented an attribute which used it but from a quick look I think it is used when attribute is applied to `ParmVar `and we want it to be inherited in redeclarations of the corresponding function. I am not sure though. Perhaps @AaronBallman is a better resource

Reply:
>> I think Attr makes sense here since we don't expect redeclaration of openCL kernel.

Agreed.